Diagnostics directly on P2P References do not display in the Dependencies node

**Steps**
Create a P2P reference from a netstandard1.3 project to a netstandard1.6 project

**Expected**
Nuget warning NU1201 in the error list, and a diagnostic message on the P2P reference in the dependencies node, with a warning icon that bubbles up.

**Actual**
NuGet warning NU1201 does appear, but the P2P node in the dependencies tree has no diagnostic or warning icon

Fixes are needed in:
- NuGet: generated diagnostic does not have a LibraryId or Target Graph (filed https://github.com/NuGet/Home/issues/5405)
- SDK: diagnostic on projects are not added to dependency node graph
- Project System: The P2P node in dependencies tree is currently only capable of showing information from the _referenced_ project, but the NuGet diagnostic comes from the _referring_ project. Previously, Project System tried to bypass NuGet/SDK and determine incompatibilities directly, but this leads to .NET Frameworks being treated as incompatible. This is fixed in https://github.com/dotnet/project-system/pull/2435

--- a/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/NuGet/Relations/ProjectToDiagnosticRelation.cs
+++ b/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/NuGet/Relations/ProjectToDiagnosticRelation.cs
@@ -1,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections;
+
+namespace Microsoft.VisualStudio.NuGet
+{
+    [Export(typeof(IRelation))]
+    internal sealed class ProjectToDiagnosticRelation : RelationBase<ProjectReferenceItem, DiagnosticItem>
+    {
+        protected override bool HasContainedItems(ProjectReferenceItem parent)
+        {
+            return parent.Target.Logs.Any(log => log.LibraryName == parent.Library.Name);
+        }
+
+        protected override void UpdateContainsCollection(ProjectReferenceItem parent, AggregateContainsRelationCollectionSpan span)
+        {
+            span.UpdateContainsItems(
+                parent.Target.Logs.Where(log => log.LibraryName == parent.Library.Name).OrderBy(log => log.LibraryName).ThenBy(log => log.Message),
+                (log, item) => StringComparer.Ordinal.Compare(log.LibraryName, item.Library.Name),
+                (log, item) => item.TryUpdateState(parent.Target, parent.Library, log),
+                log => new DiagnosticItem(parent.Target, parent.Library, log));
+        }
+
+        protected override IEnumerable<ProjectReferenceItem>? CreateContainedByItems(DiagnosticItem child)
+        {
+            yield return new ProjectReferenceItem(child.Target, child.Library);
+        }
+    }
+}

